modules/structs/hook.go

```diff
@@ -494,3 +494,17 @@ type PackagePayload struct {
 func (p *PackagePayload) JSONPayload() ([]byte, error) {
 	return json.MarshalIndent(p, "", "  ")
 }
+
+// WorkflowDispatchPayload represents a workflow dispatch payload
+type WorkflowDispatchPayload struct {
+	Workflow   string         `json:"workflow"`
+	Ref        string         `json:"ref"`
+	Inputs     map[string]any `json:"inputs"`
```

Comment:
Workflow Dispatch Payload is to transmit data to act_runner, and the `WorkflowDispatch` type in `nektos/act` uses the kv structure. [workflow.go#L77](https://github.com/nektos/act/blob/main/pkg/model/workflow.go#L77)

On the display side, to maintain the order, need to create a separate new same type for the view and use an array to store the Inputs. Because I think it will not be accepted to directly modify the kv of nektos/act into an array.
Previously discussed here: (issuecomment-1859484485)[https://github.com/go-gitea/gitea/pull/28163#issuecomment-1859484485]

in short:
* New `WorkflowDispatchInput` `WorkflowDispatch` and `workflowDispatchConfig` in `gitea/routes/web/repo/actions/view.go`, to maintain the order of inputs in forms
* Keep the kv of `WorkflowDispatchPayload` unchanged to adapt to `nektos/act`

What do you think? If you think it is feasible, I will modify it like this.